core/src/main/java/com/datastax/dse/driver/internal/core/auth/ProgrammaticDseGssApiAuthProvider.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.dse.driver.internal.core.auth;
+
+import com.datastax.dse.driver.api.core.auth.DseGssApiAuthProviderBase;
+import com.datastax.oss.driver.api.core.auth.AuthProvider;
+import com.datastax.oss.driver.api.core.metadata.EndPoint;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * {@link AuthProvider} that provides GSSAPI authenticator instances for clients to connect to DSE
+ * clusters secured with {@code DseAuthenticator}, in a programmatic way.
+ *
+ * <p>To use this provider the corresponding GssApiOptions must be passed into the provider
+ * directly, for example:
+ *
+ * <pre>
+ *    DseGssApiAuthProviderBase.GssApiOptions.builder();
+ *     Map<String, String> loginConfig =
+ *         ImmutableMap.of(
+ *             "principal",
+ *             "user principal here ex cassandra@DATASTAX.COM",
+ *             "useKeyTab",
+ *             "true",
+ *             "refreshKrb5Config",
+ *             "true",
+ *             "keyTab",
+ *             "Path to keytab file here");
+ *
+ *     builder.withLoginConfiguration(loginConfig);
+ *
+ *     CqlSession session =
+ *              CqlSession.builder()
+ *             .withAuthProvider(new ProgrammaticDseGssApiAuthProvider(builder.build()))
+ *             .build())
```

Comment:
* you need to assign the first line to a `builder` variable.
* angled brackets won't render correctly in the generated javadoc. The simplest way to address this is to replace them by `&lt;` and `&gt;`
* could you fix the indentation? Generally I create a temporary class for code examples and copy-paste them.